src/init.h

```diff
@@ -6,11 +6,15 @@
 #ifndef BITCOIN_INIT_H
 #define BITCOIN_INIT_H
 
+#include <memory>
 #include <string>
 
 class CScheduler;
 class CWallet;
 
+class WalletInitInterface;
+extern std::unique_ptr<WalletInitInterface> g_wallet_init_interface;
```

Comment:
Hmm, can we avoid yet more globals and just pass the wallet_init_interface in to the AppInit* functions?